drivers/gpio/gpio_intel_socfpga.c

```diff
@@ -0,0 +1,305 @@
+/*
+ * Copyright (C) 2022-2023 Intel Corporation
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT intel_socfpga_gpio
+
+/**
+ * @file
+ * @brief Intel GPIO SoCFPGA Controller Driver
+ *
+ * The GPIO controller on Intel SoCFPGA serves
+ * as GPIO modules. This driver provides
+ * the GPIO functionality.
+ *
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/gpio/gpio_utils.h>
+#include <zephyr/drivers/reset.h>
+
+#define LOG_LEVEL CONFIG_GPIO_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(gpio_intel_socfpga);
+
+/* Check if reset property is defined */
+#define GPIO_INTEL_INST_RESET_EN(inst)	DT_NODE_HAS_PROP(DT_DRV_INST(inst), resets)
+#define GPIO_INTEL_CHECK_RESET(inst)	GPIO_INTEL_INST_RESET_EN(inst) ||
+#define GPIO_INTEL_RESET_SUPPORT	(DT_INST_FOREACH_STATUS_OKAY(GPIO_INTEL_CHECK_RESET) 0)
+
+#define DEV_CFG(_dev) ((const struct gpio_intel_socfpga_config *)(_dev)->config)
+#define DEV_DATA(_dev) ((struct gpio_intel_socfpga_data *const)(_dev)->data)
```

Comment:
Please avoid macros like this. Just access the structure members directly.

Reply:
We are memory mapping device register using DEVICE_MMIO_NAMED_MAP macros and this macro requires DEV_DATA and DEV_CFG macros to be defined. Please check below link :

https://github.com/zephyrproject-rtos/zephyr/blob/71fd9b2e07a6fa9d7895759fc83685b6849aebb4/include/zephyr/sys/device_mmio.h#L520